src/lib/storage/resolve_encoded_column_type.hpp

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include <boost/hana/fold.hpp>
+#include <boost/hana/map.hpp>
+#include <boost/hana/pair.hpp>
+#include <boost/hana/value.hpp>
+
+#include <memory>
+
+// Include your encoded column file here!
+#include "storage/deprecated_dictionary_column.hpp"
+#include "storage/dictionary_column.hpp"
+#include "storage/run_length_column.hpp"
+
+#include "storage/encoding_type.hpp"
+
+#include "utils/enum_constant.hpp"
+#include "utils/template_type.hpp"
+
+namespace opossum {
+
+namespace hana = boost::hana;
+
+/**
+ * @brief Mapping of encoding types to columns
+ *
+ * Note: Add your encoded column class here!
+ */
+constexpr auto encoded_column_for_type = hana::make_map(
+    hana::make_pair(enum_c<EncodingType, EncodingType::DeprecatedDictionary>, template_c<DeprecatedDictionaryColumn>),
+    hana::make_pair(enum_c<EncodingType, EncodingType::Dictionary>, template_c<DictionaryColumn>),
+    hana::make_pair(enum_c<EncodingType, EncodingType::RunLength>, template_c<RunLengthColumn>));
```

Comment:
I think that I wrote this before, but we have way too many places where the possible encodings are enumerated.

Reply:
It’s true. There are three places. However, merging all of them into one file would mean including everywhere more than would be necessary.

Reply:
Including a hana map in a couple of places can't hurt too much, or does it?

Reply:
No, but these maps contains types that need to have been defined. So either we include the respective files or forward declare them.